resource/kind.go

```diff
@@ -0,0 +1,319 @@
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	_ Kind    = &UntypedKind{}
+	_ Object2 = &UntypedObject{}
+)
+
+// Object2 implements kubernetes' runtime.Object and meta/v1.Object, as well as some additional methods useful for the app-sdk
+type Object2 interface {
+	runtime.Object
+	metav1.Object
+	GetSpec() any
+	GetSubresources() map[string]any
+	GetStaticMetadata() StaticMetadata
+	SetStaticMetadata(metadata StaticMetadata)
+}
+
+// MultiVersionKind is a collection of Kinds which have the same Group and Kind, but different versions.
+// This may allow us to include conversion as a component of the interface.
+// TODO: not sure if necessary
+type MultiVersionKind interface {
```

Comment:
It would be nice to have some examples of use case that make this useful

Reply:
I don't know if you need this interface, but it would be nice to make it simple to work with each Kind at multiple versions. This would make writing a migration tool using the app-sdk more straightforward: as long as you have definitions of both versions available to you, you can use those generated types plus k8s APIs to write migration functions.

Reply:
I'm not entirely sure about the use-cases, but I agree that this might be useful for version conversion webhooks.